### ISSUE TITLE
[Bug]: Eufy login stuck at captcha screen

### ISSUE THREAD

**Issue by timmy8784** [bropat/eufy-security-client#250]
### Client version

1.5.1

### Node version

18.12.1

### Operating System type

Other

### Operating system version

Raspbian 11

### Describe the bug

Had issues with eufy security version 2.1.2 and could not login. Deleted everything and upgraded to new 2.1.3. Login takes me to captcha screen but verify spins forever until force close. Going back into plug in settings shows devices but plugin not working. Please see some logs below

### To reproduce

2. Click on '...'

### Screenshots & Logfiles

![509D582F-EA4C-4A64-BB01-D272F5125A4A](https://user-images.githubusercontent.com/59715478/202568974-b1578c65-029f-47df-8372-b3e689e50cb5.jpeg)

### Additional context

17/11/2022, 13:51:26] Homebridge v1.5.1 (HAP v0.10.4) (Homebridge 314C) is running on port 51976.
[11/17/2022, 1:51:27 PM] [EufySecurity-2.1.3] ERROR: Not connected can't continue!

When trying to login via UI, log shows the below

17/11/2022, 14:01:51] [Homebridge UI] [homebridge-eufy-security] Incoming Request: /storedAccessories
[17/11/2022, 14:01:51] [Homebridge UI] [homebridge-eufy-security] Incoming Request: /storedAccessories
[17/11/2022, 14:02:03] [Homebridge UI] [homebridge-eufy-security] Incoming Request: /login
[17/11/2022, 14:02:12] [Homebridge UI] [homebridge-eufy-security] Incoming Request: /login
[17/11/2022, 14:02:24] [Homebridge UI] [homebridge-eufy-security] /var/lib/homebridge/node_modules/homebridge-eufy-security/node_modules/eufy-security-client/build/http/station.js:107
this.rawStation.params.forEach(param => {
^

TypeError: Cannot read properties of null (reading 'forEach')
at Station.update (/var/lib/homebridge/node_modules/homebridge-eufy-security/node_modules/eufy-security-client/build/http/station.js:107:36)
at new Station (/var/lib/homebridge/node_modules/homebridge-eufy-security/node_modules/eufy-security-client/build/http/station.js:62:14)
at Station.initialize (/var/lib/homebridge/node_modules/homebridge-eufy-security/node_modules/eufy-security-client/build/http/station.js:73:25)
at EufySecurity.handleHubs (/var/lib/homebridge/node_modules/homebridge-eufy-security/node_modules/eufy-security-client/build/eufysecurity.js:360:51)
at HTTPApi. (/var/lib/homebridge/node_modules/homebridge-eufy-security/node_modules/eufy-security-client/build/eufysecurity.js:126:44)
at HTTPApi.emit (node:events:513:28)
at HTTPApi.refreshStationData (/var/lib/homebridge/node_modules/homebridge-eufy-security/node_modules/eufy-security-client/build/http/api.js:518:22)
at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
at async HTTPApi.refreshAllData (/var/lib/homebridge/node_modules/homebridge-eufy-security/node_modules/eufy-security-client/build/http/api.js:543:9)
at async EufySecurity.refreshCloudData (/var/lib/homebridge/node_modules/homebridge-eufy-security/node_modules/eufy-security-client/build/eufysecurity.js:565:9)

Node.js v18.12.1

**Comment by thieren** [comment]
Hey @timmy8784 

the error might be produced by this library, but the captcha screen and config-ui log have nothing to do with this repo.
Please provide the specific log that's produced by this library. You'll find it in the plugins data path as `eufy-log.log`

I think @bropat will need this to address the issue.

**Comment by timmy8784** [comment]
Hello and thaks for the reply!

I downloaded the logs from the plug-in and the file titled eufy-log is empty. Only the other two files configui-server and log-lib has information in it. Is there something I'm missing?

Thanks!
> Hey @timmy8784 
> 
> 
> 
> the error might be produced by this library, but the captcha screen and config-ui log have nothing to do with this repo.
> 
> Please provide the specific log that's produced by this library. You'll find it in the plugins data path as `eufy-log.log`
> 
> 
> 
> I think @bropat will need this to address the issue.

**Comment by thieren** [comment]
@timmy8784 

You'll have to [enable debug logging](https://github.com/homebridge-eufy-security/plugin/wiki/Basic-Troubleshooting#enable-logging) first and restart the plugin. After that the log file should display some messages.
The `config-ui.log` could also contain the needed information, but I think it's better to go with the plain `eufy-log.log`.

**Comment by timmy8784** [comment]
> @timmy8784 
> 
> 
> 
> You'll have to [enable debug logging](https://github.com/homebridge-eufy-security/plugin/wiki/Basic-Troubleshooting#enable-logging) first and restart the plugin. After that the log file should display some messages.
> 
> The `config-ui.log` could also contain the needed information, but I think it's better to go with the plain `eufy-log.log`.

Ah ok got it now!

Here is the eufy log. Thanks!

2022-11-19 17:27:10.053  DEBUG [undefined.constructor] Loaded API_BASE: https://security-app.eufylife.com 
2022-11-19 17:27:10.260  DEBUG [EufySecurity._initializeInternals] Load previous login_hash: 311128a3ab878d27e98b4f68914aa64e 
2022-11-19 17:27:11.032  DEBUG [HTTPApi.login] Login and get an access token 
{
  token: null,
  tokenExpiration: null
}
2022-11-19 17:27:11.166  ERROR [HTTPApi.login] Generic Error: 
 TypeError  The "data" argument must be of type string or an instance of Buffer, TypedArray, or DataView. Received undefined
details:
{
  code: 'ERR_INVALID_ARG_TYPE'
}
error stack:
• errors:393 NodeError
    node:internal/errors:393:5

• cipher:176 update
    node:internal/crypto/cipher:176:11

• utils.ts:139 encryptPassword
    node_modules/homebridge-eufy-security/node_modules/eufy-security-client/src/http/utils.ts:139:16

• api.ts:269 login
    node_modules/homebridge-eufy-security/node_modules/eufy-security-client/src/http/api.ts:269:47

• eufysecurity.ts:673 connect
    node_modules/homebridge-eufy-security/node_modules/eufy-security-client/src/eufysecurity.ts:673:24

• platform.ts:209 pluginSetup
    node_modules/homebridge-eufy-security/src/plugin/platform.ts:209:29

• platform.ts:176 <anonymous>
    node_modules/homebridge-eufy-security/src/plugin/platform.ts:176:7

**Comment by Palmke** [comment]
What is your configuration? What is the value of `trustedDeviceName`?

**Comment by thieren** [comment]
Hi @Palmke 

According to the description he uses the plugin version `2.1.3` which itself is running on eufy-security-client `2.2.3` (so the latest).

If I'm not mistaken if the `trustedDeviceName` is left empty the new version will choose one at random, right?
But @timmy8784 can himself say if he has changed that value.

**Comment by Palmke** [comment]
Yes but only if it is not set already. The error looks to be the same I reported last week.

**Comment by thieren** [comment]
Ah, now I'm with you.

Deleting the `persistent.json` should do the trick, if this is indeed already set to the old value, right?

@timmy8784 can you do that as a test? Deleting the `persistent.json` and try login in again. (In the plugin config is a button `Remove everything` which should do the trick, but double check for good measure.)

**Comment by bropat** [comment]
@timmy8784 

The first error was fixed with #246.

@thieren 

The `trustedDeviceName` parameter is a setting of the driver and is not stored in the `persistent.json` file.

> If I'm not mistaken if the trustedDeviceName is left empty the new version will choose one at random, right?

Only if not set. An empty string is still accepted in the current version and can lead to login problems. In the next version (`2.3.0`), an empty string will also lead to the fallback mechanism.

When the fallback mechanism has taken effect, the parameter `fallbackTrustedDeviceName` is set in the `persistent.json` file.
Here you can then see the randomly selected value.

**Comment by bropat** [comment]
In this case, I expect that there is no `fallbackTrustedDeviceName` parameter in the `persistent.json` file.
Otherwise, the above error should not occur.

**Comment by timmy8784** [comment]
> Ah, now I'm with you.
> 
> 
> 
> Deleting the `persistent.json` should do the trick, if this is indeed already set to the old value, right?
> 
> 
> 
> @timmy8784 can you do that as a test? Deleting the `persistent.json` and try login in again. (In the plugin config is a button `Remove everything` which should do the trick, but double check for good measure.)

I have done the "remove everything" at least a dozen times or so and have the same issue. I've also tried adding a name in the device name field and also leaving it blank. No matter what I try I still run into the same issue

**Comment by timmy8784** [comment]
I tried to reset again and still same issues. I went to plug-in settings and did the remove everything option. I then deleted the plug-in and reinstalled the latest 2.1.3 version and still gets stuck at login.

**Comment by timmy8784** [comment]
Any update on this?

**Comment by jazzdan** [comment]
I bought a Eufy doorbell camera a month ago and I have always had this problem, until today.

When I had this infinite captcha problem I was trying to log in as a user I had created specifically for eufy, and not a user that I had originally used to setup my account.

When I switched to the user that I had used to originally create my Eufy account instead of getting a CAPTCHA image I got an authentication code sent to my email, which worked and allowed me to authenticate the homebridge plugin.
